app/params/builder.go

```diff
@@ -0,0 +1,16 @@
+package params
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	buildertypes "github.com/skip-mev/pob/x/builder/types"
+)
+
+func SetBuilderDefaultConfig() {
+	sdk.SetAddrCacheEnabled(false)
+	defer sdk.SetAddrCacheEnabled(true)
```

Comment:
**Q:** what does it do?

**A:**
Check out the line below where module address converted to string via `.String()`  - see the implementation
it has internal global cache

Basically after setting config to have addresses prefix changed from `cosmos1`  to `persistence1`, it still had cached `cosmos1` only if cache is not disabled

....

I am not sure if this is a problem, but most old chains use config override inside init()  so it's applied as soon as package is included. But in our code we set config in the beginning of a root cmd, so I guess things are already happening before that point